src/execution/isolate.cc

```diff
@@ -5384,9 +5389,16 @@ void Isolate::RunAllPromiseHooks(PromiseHookType type,
   }
 }
 
+extern void RecordReplayOnPromiseHook(Isolate* isolate, PromiseHookType type,
+                                      Handle<JSPromise> promise, Handle<Object> parent);
+
 void Isolate::RunPromiseHook(PromiseHookType type, Handle<JSPromise> promise,
                              Handle<Object> parent) {
   if (!HasIsolatePromiseHooks()) return;
+  if (recordreplay::IsReplaying()) {
+    RecordReplayOnPromiseHook(this, type, promise, parent);
```

Comment:
Missing an `AutoDisallowEvents`?

Reply:
I don't think that's necessary here, RecordReplayOnPromiseHook is not a complicated function and doesn't interact with the recording or system.